ergo-chain-sync/src/cache/rocksdb.rs

```diff
@@ -6,75 +6,179 @@ use ergo_lib::{
+                        id: oldest_id,
+                        height: oldest_height,
+                    } = bincode::deserialize(&bytes).unwrap();
+
+                    // Replace OLDEST_BLOCK if the persistent store is at capacity.
+                    if block.height - oldest_height == max_rollback_depth {
+                        let new_oldest_block = BlockRecord {
+                            id: bincode::deserialize(
+                                &db_tx
+                                    .get(&postfixed_key(&oldest_id, CHILD_POSTFIX))
+                                    .unwrap()
+                                    .unwrap(),
+                            )
+                            .unwrap(),
+                            height: oldest_height + 1,
+                        };
+                        db_tx
+                            .put(
+                                bincode::serialize(OLDEST_BLOCK).unwrap(),
+                                bincode::serialize(&new_oldest_block).unwrap(),
+                            )
+                            .unwrap();
+
+                        // Delete all data relating to the 'old' oldest block
+                        let tx_ids_bytes = db_tx
+                            .get(&postfixed_key(&oldest_id, TRANSACTION_POSTFIX))
+                            .unwrap()
+                            .unwrap();
+                        let tx_ids: Vec<TxId> = bincode::deserialize(&tx_ids_bytes).unwrap();
+                        for tx_id in tx_ids {
+                            let tx_key = bincode::serialize(&tx_id).unwrap();
+                            db_tx.delete(tx_key).unwrap();
+                        }
+
+                        db_tx
+                            .delete(&postfixed_key(&oldest_id, TRANSACTION_POSTFIX))
+                            .unwrap();
+                        db_tx.delete(&postfixed_key(&oldest_id, HEIGHT_POSTFIX)).unwrap();
+                        db_tx.delete(&postfixed_key(&oldest_id, PARENT_POSTFIX)).unwrap();
+                        db_tx.delete(&postfixed_key(&oldest_id, CHILD_POSTFIX)).unwrap();
+                    }
+                } else {
+                    // This is the very first block to add to the store
+                    db_tx
+                        .put(
+                            bincode::serialize(OLDEST_BLOCK).unwrap(),
+                            bincode::serialize(&BlockRecord {
+                                id: block.id,
+                                height: block.height,
+                            })
+                            .unwrap(),
+                        )
+                        .unwrap();
+                }
+
+                match db_tx.commit() {
+                    Ok(_) => {
+                        break;
+                    }
+                    Err(e) => {
+                        if e.kind() == rocksdb::ErrorKind::Busy {
+                            continue;
```

Comment:
I guess this part can be omitted as ChainCache instance is supposed to use its rocksdb instance exclusively, so concurrent transactions should not be possible.

Reply:
Sure, removed the loop.